fatrace.c

```diff
@@ -225,15 +228,73 @@ show_pid (pid_t pid)
     return true;
 }
 
+static bool
+is_valid_string(const char *str)
```

Comment:
Please rename to `is_valid_utf8()`, which is more self-descriptive.

Reply:
Hmm, what does that mean though? In e.g. a latin1 locale, will it try to interpret the bytes as ISO-8859-1 and convert to UCS32 that way? I.e. totally mess up?

I *do* think we want/need an `is_valid_utf8()` -- after all, you said yourself that JSON is specified to be UTF-8?

If we can't have that, then maybe restrict to ASCII at least in the first iteration?

Reply:
Excellent suggestion.